libraries/hwlib/library/targets/hwlib-arduino-due.hpp

```diff
@@ -861,6 +895,172 @@ class pin_adc : public hwlib::adc {
    }
 };
 
+class usart_bus {
+private:
+//    Pio & tx_port;
+//    Pio & rx_port;
+//    uint32_t tx_mask;
+//    uint32_t rx_mask;
+
+    Usart * hardwareUSART = nullptr;
+    unsigned int baudrate = BMPTK_BAUDRATE;
+    int n;
+    bool USART_controller_initialized = false;
+
+    Queue<uint8_t, 250> input_buffer;
+   
+    inline uint8_t receive_byte(){
+        return hardwareUSART->US_RHR;
+    }
+
+    unsigned int available() {
+        if (!USART_controller_initialized) {
+            return 0;
+        }
+
+        if ((hardwareUSART->US_CSR & 1) != 0) {
+            input_buffer.push(receive_byte());
+        
+        }
+        return input_buffer.count();
+    }
+
+    void send_byte(const uint8_t &b){
+        while (!tx_ready());
+        hardwareUSART->US_THR = b;
+    }
+    inline bool tx_ready(){
+        return (hardwareUSART->US_CSR & 2);
+    }
+
+public:
+    usart_bus(
+//        uint32_t tx_port_number, 
```

Comment:
why are there random comments in here? They don't explain me anything

Reply:
I see, I forgot to uncomment some code